python/surf/misc/_McsReader.py

```diff
@@ -37,31 +38,36 @@ def __init__(self,name="McsReader"):
         self.size      = 0
         
     def open(self, filename):   
-        self.entry     = []
         self.startAddr = 0
         self.endAddr   = 0
         self.size      = 0
         baseAddr       = 0
-        dataList       = []
-
+        idx            = 0
+        
         # Check for non-compressed .MCS file
         if fnmatch.fnmatch(filename, '*.mcs'):
+            # Set the flag
             gzipEn = False
+            # Find the length of the file
+            numLines = int(subprocess.check_output('wc -l {}'.format(filename), shell=True).split()[0])
+            
+        # Check for Compressed .MCS file
         elif fnmatch.fnmatch(filename, '*.mcs.gz'):
+            # Set the flag
             gzipEn = True
+            # Find the length of the file
+            numLines = sum(1 for line in gzip.open(filename, "rb"))
```

Comment:
Wouldn't this also read the whole file line by line like before?

Reply:
I have no choice for the .gz file version.  90% of the time we are dealing with non-compressed file so this seem like a reasonable compromise.  
Do you have a better method for getting the number of lines inside a .gz file?

Reply:
That is true... Maybe you could `zcat file.gz | wc -l` ?
It would extract to a stream in memory and never touch the disk...

Reply:
Applied your recommendation in 4173ee5